.github/workflows/lighthouse-ci.yml

```diff
@@ -0,0 +1,69 @@
+name: Lighthouse CI
+
+on:
+  pull_request:
+    branches:
+      - master
+    types: [opened, reopened, synchronize, ready_for_review]
+
+jobs:
+  lighthouse-ci:
+    name: Lighthouse CI
+    runs-on: ubuntu-latest
+    if: "github.repository == 'asyncapi/website' && github.event.pull_request.draft == false &&!((github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'ci: update global workflows')) || (github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'chore(release):')) || (github.actor == 'allcontributors' && startsWith(github.event.pull_request.title, 'docs: add')))"
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Await Netlify Preview
+        uses: jakepartusch/wait-for-netlify-action@v1
+        id: netlify
+        with:
+          site_name: asyncapi-website
+          max_timeout: 300
+
+      - name: Lighthouse Audit
+        id: lighthouse_audit
+        uses: treosh/lighthouse-ci-action@9.3.0
+        with:
+          urls: |
+            https://deploy-preview-$PR_NUMBER--asyncapi-website.netlify.app/
+          configPath: ./.github/workflows/lighthouserc.json
+          uploadArtifacts: true
+          temporaryPublicStorage: true
+        env:
+          PR_NUMBER: ${{ github.event.pull_request.number}}
+
+      - name: Lighthouse Score Report
+        id: lighthouse_score_report
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
what is token needed here for?

Reply:
According to the [docs](https://github.com/actions/github-script#examples), the `GITHUB_TOKEN` is optional in this action. However, I have added it so that the workflow can access it intrinsically to put up a comment on the PR.